### DISCUSSION TITLE
attach / detach

### DISCUSSION THREAD

**Discussion by teho1** [cutelyst/cutelyst#326] [Q&A]
I'm using the Grantlee view and trying to read data asynchronously and stash the data to the view on the slot as follows, but I could not get it working.
```
void Calendar::calendar(Context *c, const QString &name)
{
    QTimer *timer = new QTimer(this);
    timer->start(1000);
    connect(timer, &QTimer::timeout, c, [=] {
    QLocale locale("EN");
    QVariantList daynames;
    for (int i=1; i<8; i++) {
        QString day = locale.dayName(i, QLocale::NarrowFormat);
        daynames.append(day);
    }
    c->stash({
                {"daynames", daynames},
                {"template", "calendar.html"}
    });
    c->attachAsync();
    });

    c->detachAsync();
}
}
```
Without attachAsync / detachAsync I'm getting the following output, but with async, the daynames remains empty:
```
<ul class="weekdays">
  <li>M</li>
  <li>T</li>
  <li>W</li>
  <li>T</li>
  <li>F</li>
  <li>S</li>
  <li>S</li>
</ul>
</body>
</html>
```

**Comment by dantti** [comment]
Why it doesn't work? Also depending on the Cutelyst version you will want to use the ASync class to make this code simpler.

**Reply by teho1** [reply]
Thanks for the reply. I'm using the latest code from the master branch. I edited the example code and the output. Maybe I'm using it the wrong way? I'll try the ASync class next.

**Reply by dantti** [reply]
I just tried your code and it works, the only difference is that I'm using Cutelee instead of Grantlee but I can't see that as a problem, are you using Path or Chained dispatch?

**Reply by teho1** [reply]
I also switched to Cutelee (6.0.0). I have following declarations:

C_ATTR(calendar, :Local :Args(1) :ActionClass("RenderView") :View("cutelee_view"))
void calendar(Context *c, const QString &name);

**Reply by teho1** [reply]
Do you mind sharing your code as mine is still not working?

**Reply by dantti** [reply]
oh, ok your issue is your ActionClass, ActionClasses call your method directly and RenderView doesn't know about async so it will render the view right after the method was called, I'll see if I can make it aware but for you can either have RenderView at End() method or call c->view()->render();
BTW you can replace :Args(1) with :AutoArgs()

**Reply by dantti** [reply]
Took a look at RenderView and I can't change it because I can't call the same action when I attach async, so you have to do one of those two options.

**Reply by teho1** [reply]
Thanks a lot! I moved the RenderView to the End and it works fine now.
